### PR TITLE
Add arch linux build instructions

### PR DESCRIPTION
**Pull request by brendankristiansen**
## Description

Provides instructions for quickly building QGIS on an Arch-based workstation

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -479,6 +482,34 @@ default.debug=true
...
+### 3.12.1. Install Build Dependencies
+
+```bash
+sudo pacman -S pyqt-builder gsl gpsbabel postgresql sip libzip qtkeychain-qt5 qca-qt5 qscintilla-qt5 expat qwt gdal \
+spatialindex sqlite geos proj qt5 libspatialite cmake gcc flex bison
+```
+
+### 3.12.2. Compiling
+
+In the root of the repsitory, at any tag or branch you choose:
```
Comment (DelazJ) [inline INSTALL.md:496]:
```suggestion
In the root of the repository, at any tag or branch you choose:
```

File: INSTALL.md [hunk 2]
```diff
@@ -479,6 +482,34 @@ default.debug=true
...
 ```
 
+## 3.12. Building on Arch Linux
+
+### 3.12.1. Install Build Dependencies
+
+```bash
+sudo pacman -S pyqt-builder gsl gpsbabel postgresql sip libzip qtkeychain-qt5 qca-qt5 qscintilla-qt5 expat qwt gdal \
+spatialindex sqlite geos proj qt5 libspatialite cmake gcc flex bison
+```
```
Comment (pblottiere) [inline INSTALL.md:492]: Installing the whole `qt5` package comes with a lot of unnecessary packages for QGIS:

``` bash
:: There are 40 members in group qt5:
:: Repository extra
   1) pyside2  2) qt5-3d  3) qt5-base  4) qt5-charts  5) qt5-connectivity  6) qt5-datavis3d  7) qt5-declarative  8) qt5-doc  9) qt5-examples  10) qt5-gamepad  11) qt5-graphicaleffects  12) qt5-imageformats
   13) qt5-location  14) qt5-lottie  15) qt5-multimedia  16) qt5-networkauth  17) qt5-purchasing  18) qt5-quick3d  19) qt5-quickcontrols  20) qt5-quickcontrols2  21) qt5-quicktimeline  22) qt5-remoteobjects
   23) qt5-script  24) qt5-scxml  25) qt5-sensors  26) qt5-serialbus  27) qt5-serialport  28) qt5-speech  29) qt5-svg  30) qt5-tools  31) qt5-translations  32) qt5-virtualkeyboard  33) qt5-wayland
   34) qt5-webchannel  35) qt5-webengine  36) qt5-webglplugin  37) qt5-websockets  38) qt5-webview  39) qt5-x11extras  40) qt5-xmlpatterns
``` 

While it's of course the most simple solution, I don't know the current policy for the `INSTALL.md` file: simplicity or lightest compilation environment?

Comment (m-kuhn) [reply INSTALL.md:492]: looking at debian build instructions https://github.com/qgis/QGIS/blob/7d3b1958e0ae6358dd0bb093e1eba53f3939ffb1/INSTALL.md#33-install-build-dependencies I'd rather suspect light

Comment (pblottiere) [reply INSTALL.md:492]: Hi @brendankristiansen,

> looking at debian build instructions https://github.com/qgis/QGIS/blob/7d3b1958e0ae6358dd0bb093e1eba53f3939ffb1/INSTALL.md#33-install-build-dependencies I'd rather suspect light

Can you update this PR accordingly please?

File: INSTALL.md [hunk 3]
```diff
@@ -479,6 +482,34 @@ default.debug=true
...
+
+In the root of the repository, at any tag or branch you choose:
+
+Keep in mind, this will destroy any contents of your build directory if you have one.
+
+```bash
+if [ -d ./build/ ]; then rm -rfv ./build/*; else mkdir -v build; fi
+cd build
+cmake ..
+make
```
Comment (kannes) [inline INSTALL.md:504]: `cmake --build .` will use all cores. :>

File: INSTALL.md (hunk 2 above)
Comment (kannes) [reply INSTALL.md:492]: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=qgis-git might be another reference

File: INSTALL.md (hunk 3 above)
Comment (m-kuhn) [reply INSTALL.md:504]: Is this something that should be documented per distro?
One single reference would be easier to keep up to date and could also go more in depth (e.g. explaining `CMAKE_BUILD_TPYE`)

My favorite

```
cmake -GNinja -S . -B build
cmake --build build
```

File: INSTALL.md [hunk 4]
```diff
@@ -479,6 +482,34 @@ default.debug=true
...
+```
+
+Optionally:
+
+```bash
+sudo make install
```
Comment (m-kuhn) [inline INSTALL.md:510]: This should have some context.
There are other methods to run a self-compiled qgis (directly from the build dir or installing into a user directory) which I prefer in almost any case. Those who are at ease with system administration will know that this is an option even without this documentation. Those who need this documentation (and will just copy paste it) are probably not aware that it's not that easy to undo.

**Pull request closed without merging**